### PR TITLE
re-introduce send-side streaming

### LINKED ISSUE #14: Send-side streaming
Currently we're buffering the entire file in browser memory on both send and receive sides before/after streaming to/from the relay, respectively. This issue focuses only on the receive-side.

In the send case, I think our memory footprint is double the file size because the browser buffers the entire file into memory and then we copy that buffer into the WASM module. We can improve on this by streaming the file data between the JavaScript and the WASM module.

### PR DESCRIPTION
**Pull request by vu3rdd**
Closes #14 

## Code Review Checklist (to be filled out by reviewer)

- [ ] Description accurately reflects what changes are being made.
- [ ] Description explains why the changes are being made (or references an issue containing one).
- [ ] The PR appropriately sized.
- [ ] New code has enough tests.
- [ ] New code has enough documentation to answer "how do I use it?" and "what does it do?".
- [ ] Existing documentation is up-to-date, if impacted.

### CODE REVIEW

**Comment by vu3rdd** [comment]
I tested it with a slightly older commit in `main` along with https://github.com/LeastAuthority/wormhole-william/pull/39 and it worked fine. But when rebased against `main` (which is what this branch is), the screen showing the code (while sending) doesn't appear after selecting the file. Will rebase this branch once the `main` is fixed.

**Comment by vu3rdd** [comment]
The `streaming` tests are crashing because of the interface of `NewTestFile()`, so converting this to draft for now. Will work on improving that part.

File: src/go/wormhole/client.ts [hunk 1]
```diff
@@ -22,8 +22,8 @@ export default class Client implements ClientInterface {
     }
 
     public async sendFile(file: File, opts?: TransferOptions): Promise<TransferProgress> {
-        const data = new Uint8Array(await file.arrayBuffer());
-        return wormhole.Client.sendFile(this.goClient, file.name, data, opts);
+        // XXX get rid of file.name param to sendFile()?
```
Comment (ewanas) [inline src/go/wormhole/client.ts:25]: Looks like XXX could be TODO, otherwise I agree, `file.name` is redundant.

**Review by ewanas — DISMISSED** [review]
LGTM. Probably wormhole-william will need to be updated in this PR after https://github.com/LeastAuthority/wormhole-william/pull/39 is merged.

**Comment by vu3rdd** [comment]
> LGTM. Probably wormhole-william will need to be updated in this PR after [LeastAuthority/wormhole-william#39](https://github.com/LeastAuthority/wormhole-william/pull/39) is merged.

Thanks @ewanas. I just updated the submodule and pushed that commit. Will wait for the CI to run.

**Review by ewanas — APPROVED** [review]

**Pull request merged**